python/lsst/pipe/base/pipelineIR.py

```diff
@@ -637,6 +668,28 @@ def _read_labeled_subsets(self, loaded_yaml: dict[str, Any]) -> None:
         for key, value in loaded_subsets.items():
             self.labeled_subsets[key] = LabeledSubset.from_primitives(key, value)
 
+    def _read_step_declaration(self, loaded_yaml: dict[str, Any]) -> None:
+        """Process the steps portion of the loaded yaml document
+
+        Steps are subsets that are declared to be normal parts of the overall
+        processing of the pipeline. Not all subsets need to be a step, as they
+        can exist for certain targeted processing, such as debugging.
+
+        Parameters
+        ----------
+        loaded_yaml: `MutableMapping`
```

Comment:
```suggestion
        loaded_yaml: `dict`
```

If the type annotation says `dict`, use that here.  If you want `MutableMapping` in both places, you have to say `~collections.abc.MutableMapping` in the docs for it to link properly.